src/sys/socket/sockopt.rs

```diff
@@ -1015,6 +1039,50 @@ sockopt_impl!(
     libc::IP_TTL,
     libc::c_int
 );
+#[cfg(any(apple_targets, linux_android))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    /// Retrieve the current time-to-live field for every
+    /// IPv4 packet received on this socket.
+    IpRecvTtl,
+    Both,
+    libc::IPPROTO_IP,
+    libc::IP_RECVTTL,
+    bool
+);
+#[cfg(any(apple_targets, linux_android, target_os = "freebsd",))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    /// Retrieve the current time-to-live field for every
+    /// IPv4 packet received on this socket.
+    IpTtl,
+    Both,
+    libc::IPPROTO_IP,
+    libc::IP_TTL,
+    libc::c_int
+);
```

Comment:
There is an `Ipv4Ttl` above this, and they seem to be the same thing?

Reply:
Yes, that was there already, but the `Ipv4Ttl` name is unfortunate, because it's not obvious that this is the socket option that should be used for per-packet CMSG use. What do you prefer I do?

Reply:
Ahh, they are the same thing under the hood, but for different usages, then this is a really unfortunate naming, do you mind give them some documents on their typical use cases and how they should be properly used, some examples would be great!